File: 16-express-js-hexa/adapters/web/dtos/userDTO.js
import {User} from '../../../domain/models/user.js'

export class UserDTO {
    constructor(id, name, email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }

    static fromDomain(user) {
        return new UserDTO(
            user.id,
            user.name,
            user.email
        );
    }
}

File: 16b-fastapi-python-hexa/adapters/web/dtos/user_dto.py
from domain.models.user import User
from pydantic import BaseModel, EmailStr

class UserDTO(BaseModel):
    id: str
    name: str
    email: EmailStr

    @classmethod
    def from_domain(cls, user: User) -> 'UserDTO':
        return cls(
            id=user.id,
            name=user.name,
            email=user.email
        )
